crates/plugins/src/manifest.rs

```diff
@@ -88,23 +88,32 @@ impl Architecture {
 }
 
 /// Checks whether the plugin supports the currently running version of Spin.
-pub fn check_supported_version(manifest: &PluginManifest, spin_version: &str) -> Result<()> {
+pub fn check_supported_version(
+    manifest: &PluginManifest,
+    spin_version: &str,
+    override_compatibility_check: bool,
+) -> Result<()> {
     let supported_on = &manifest.spin_compatibility;
-    inner_check_supported_version(supported_on, spin_version)
+    inner_check_supported_version(supported_on, spin_version, override_compatibility_check)
 }
 
-fn inner_check_supported_version(supported_on: &str, spin_version: &str) -> Result<()> {
+fn inner_check_supported_version(
+    supported_on: &str,
+    spin_version: &str,
+    override_compatibility_check: bool,
+) -> Result<()> {
     let comparator = VersionReq::parse(supported_on).with_context(|| {
-        format!(
-            "Could not parse manifest compatibility version {} as valid semver",
-            &supported_on,
-        )
+        format!("Could not parse manifest compatibility version {supported_on} as valid semver")
```

Comment:
Suggestion, though probably an irrelevant one... if `override_compatibility_check` is set, a parse error should _not_ be fatal (the user has already said they don't care about versions).  That said, it's hard to imagine how any plugin with a broken version spec gets this far!